tests/pytests/test_geometry_flat.py

```diff
@@ -110,6 +110,27 @@ def testSanitySearchHashIntersectsDisjoint(env):
   res = env.cmd('FT.SEARCH', 'idx', '@geom:[disjoint $poly]', 'PARAMS', 2, 'poly', query, 'NOCONTENT', 'DIALECT', 3)
   env.assertEqual(toSortedFlatList(res), [2, 'tall', 'wide'])
 
+def test_MOD_7126(env):
+  conn = getConnectionByEnv(env)
+  env.expect('FT.CREATE', 'idx', 'SCHEMA', 'geom', 'GEOSHAPE', 'FLAT').ok()
+
+  point1 = 'POINT(10 10)'
+  point2 = 'POINT(50 50)'
+  triangle = 'POLYGON((20 20, 25 35, 35 25, 20 20))'
+  rectangle = 'POLYGON((60 60, 65 75, 70 70, 65 55, 60 60))'
```

Comment:
Can add an intersecting shape which is not within the `query` shape?
```suggestion
  rectangle = 'POLYGON((60 60, 65 75, 70 70, 65 55, 60 60))'
  rectangle2 = 'POLYGON((60 10, 60 20, 70 20, 70 10, 60 10))'
```

Reply:
![plot](https://github.com/RediSearch/RediSearch/assets/76992134/777b941b-3f1d-44cc-96d7-1d513cb4ff24)

Reply:
`testSanitySearchHashIntersectsDisjoint` already covers such cases